WC_Layers/Source_Mesh_Reader.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 30 15:16:44 2024
+
+@author: Anupama Rajendra
+"""
+
+import h5py
+import numpy as np
+
+source_density_name = 'source_density'
+
+# Open the HDF5 files
+File_1 = h5py.File("source_mesh_1.h5m", 'r')
+File_2 = h5py.File("source_mesh_2.h5m", 'r')
+Files = [File_1, File_2]
+
+def extract_source_data(file_list):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
+    sd_list = []

```

Comment:
If there is a way to know the size of this data structure ahead of time, I recommend making this a numpy array that you fill with values below.  For example:
```
# assume we know num_mesh_elements, num_groups
sd_list = np.ndarray( (len(file_list), num_mesh_elements, num_groups) )

for file_num, file in enumerate(file_list):
    sd_list[:, file_num] =  file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```